src/components/common.js

```diff
@@ -303,9 +303,10 @@ const commonSelectProps = {
  * @param {Object} props - see {@link https://react-select.com/props#select-props}
  * @param props.value - a member of options
  * @param {Array} props.options - can be of any type; if objects, they should each contain a value and label, unless defining getOptionLabel
+ * @param props.withGroups - boolean; see {@link https://react-select.com/advanced#replacing-builtins} for info on formatting the options array
  */
-export const Select = ({ value, options, id, ...props }) => {
-  const newOptions = options && !_.isObject(options[0]) ? _.map(value => ({ value }), options) : options
+export const Select = ({ value, options, id, withGroups = false, ...props }) => {
```

Comment:
Looking at the underlying `react-select` code, it detects groups by looking for the presence of an `options` attribute on the items, so we could probably do the same thing (instead of passing an explicit flag).

Reply:
Checking the presence of an `options` item in the object (as far as I could figure out) required all these checks to ensure the object existed `newOptions && newOptions[0] && newOptions[0].options` so to me it felt like passing an explicit flag would be simpler/clearer? However I might not be not following your suggestion correctly. Please let me know if you still think I should try for this approach rather than the flag option based on my latest changes. thanks!

Reply:
It seems like the crux of the change is in findValue. When using a grouped select we need to flatten the options.
Another approach would be to make a property around how to treat options when finding an item, by default it would be a no-op. 

Or we could make an unexposed "BaseSelect" component that accepts a findFn and wrap it with other components  (Select and GroupedSelect) that pass in the find function as a prop. This would keep the component props the same for both select types, not exposing additional complexity when using the components and removing the conditional/branching logic for grouping within the component.

Reply:
Please see latest changes: I have updated the solution to rely on a "`baseSelect`" with "`groupedSelect`" and "`Select`" wrappers to pass a `findValue` function.